docs/ExpectAPI.md

```diff
@@ -495,6 +499,177 @@ it('transitions as expected', () => {
+import {expect, test} from '@jest/globals';
+import {Volume} from '../Volume.js';
+import '../areVolumesEqual.js';
+
+test('are equal with different units', () => {
+  expect(new Volume(1, 'L')).toEqual(new Volume(1000, 'mL'));
+});
+```
+
+#### Custom equality testers API
+
+Custom testers are functions that return either the result (`true` or `false`) of comparing the equality of the two given arguments or `undefined` if tester does not handle the given the objects and wants to delegate equality to other testers (for example, the built in equality testers).
+
+Custom testers are called with 3 arguments: the two objects to compare and the array of custom testers (used for recursive testers, see section below).
+
+#### Matchers vs Testers
+
+Matchers are methods available on `expect`, for example `expect().toEqual()`. `toEqual` is a matcher. A tester is a method used by matchers that do equality checks to determine if objects are the same.
+
+Custom matchers are good to use when you want to provide a custom assertion that test authors can use in their tests. For example, the `toBeWithinRange` example in the `expect.extend` section is a good example of a custom matcher. Sometimes a test author may want to assert two numbers are exactly equal and should use `toBe`. Other times however, a test author may want to allow for some flexibility in their test and `toBeWithinRange` may be a more appropriate assertion.
+
+Custom equality testers are good to use for globally extending Jest matchers to apply custom equality logic for all equality comparisons. Test authors can't turn on custom testers for certain assertions and turn off for others (a custom matcher should be used instead if that behavior is desired). For example, defining how to check if two `Volume` objects are equal for all matchers would be a good custom equality tester.
+
+#### Recursive custom equality testers
+
+If you custom equality testers is testing objects with properties you'd like to do deep equality with, you should use the `equals` helper from the `@jest/expect-utils` package. This `equals` method is the same deep equals method Jest uses internally for all of its deep equality comparisons. Its the method that invokes your custom equality tester. It accepts an array of custom equality testers as a third argument. Custom equality testers are also given an array of custom testers as their third argument. Pass this argument into the third argument of `equals` so that any further equality checks deeper in your object can also take advantage of custom equality testers.
+
+For example, let's say you have a `Book` class that contains an array of `Author` classes and both of these classes have custom testers. The `Book` custom tester would want to do a deep equality check on the array of `Author`s and pass in the custom testers so the `Author`s custom equality tester is applied:
+
+```js title="customEqualityTesters.js"
+import {equals} from '@jest/expect-utils';
```

Comment:
can we ask people to use `this.equals` instead of importing it?

Reply:
right, that makes sense! Yeah, I think we should have a context with at least `equals` (maybe only that for now?). That way, if people publish custom equality testers, they don't need to care what version of Jest people are using (beyond 29.4+) when we release 30, 31 etc..

Reply:
Oh nice! Yea, that makes sense. I believe the latest changes I've pushed do this. Let me know if you see anything else missing!

I've also re-exported the `Tester` and `TesterContext` types from the `expect` package. My thinking was that since those types are implicitly exposed through `expect.addEqualityTester`, explicitly exporting them would make using TypeScript a little easier and better mirror what people can do with `expect.extend` with the `MatcherFunction` and `MatcherContext` types. However, if you'd prefer people import `Tester` and `TesterContext` directly from `@jest/expect-utils`, we can simply revert that commit.

Reply:
sounds good 👍